### DISCUSSION TITLE
Invoice number format is not correct per customer

### DISCUSSION THREAD

**Discussion by chrisonline** [kimai/kimai#3824] [General]
### Describe the issue

I have set following invoice numer format:
![image](https://user-images.githubusercontent.com/2077218/216921486-64038828-8329-464a-9c39-8c961a33b637.png)

`ccy` should mean as of the docs:
{ccy} - per customer invoice counter for this year, starting at 1

But I have now created 2 invoices, one for each customer, but I get for the first invoice 2023/001 and for the second invoice from a different customer 2023/002, but it should be 2023/001 again.

The invoice template I use is set to this:
![image](https://user-images.githubusercontent.com/2077218/216921988-ef81ff9a-a0e1-4986-854c-58802c7e4447.png)

### I already tried

- [X] I've read and searched [the documentation](https://www.kimai.org/documentation/).
- [X] I've searched for similar issues in this repository.
- [X] I've searched for similar issues in [the discussions](https://github.com/kimai/kimai/discussions).

### Kimai version

1.30.1 stable

### How do you run Kimai?

Synology

### Which PHP version are you using?

8.1

**Comment by kevinpapst** [comment]
This replacer only makes sense in combination with something else. 
Kimai will NEVER create the same invoice number twice.
See the warning box at https://www.kimai.org/documentation/invoices.html#invoice-number-format (which I just updated to explain that fact).

**Comment by chrisonline** [comment]
Ok thanks makes sense :-)
So I need to have for example {Y}/{cnumber}{ccy,3} and it should work?

**Reply by kevinpapst** [reply]
Yes, that is a logical combination that should work and that might make your accountant happy as well.

**Comment by bliepp** [comment]
So, wouldn't it be great to have an `{cid}` option (just like `{uid}` but for the customer's ID)? Not every customer might have a customer number (hence using `{cnumber}`might throw an error) and a customer name might be subject to change over time. As far as I'm concerned a truly unique identifier that's guaranteed to exist (i.e. the customer's ID) would be very useful for consistent per-customer invoice numbering. Sure, `{cname}`should also work, but a) it might be considered inconsistent when the customer's name changes and b) having the customers name appear twice in the file just doesn't look that nice.
